test/integ/operator_integ_test.go

```diff
@@ -0,0 +1,158 @@
+			operatorAssertions:       verifyOperator,
+			agentAssertions:          verifyAgent,
+			installCertManager:       false,
+		},
+		{
+			name: "Datadog agent with Operator Helm install, conversion webhook enabled",
+			command: common.HelmCommand{
+				ReleaseName: "datadog-operator-with-webhook",
+				ChartPath:   "../../charts/datadog-operator",
+				Overrides: map[string]string{
+					"installCRDs": "true",
+					"datadogCRDs.migration.datadogAgents.version":                   "v2alpha1",
+					"datadogCRDs.migration.datadogAgents.useCertManager":            "true",
+					"datadogCRDs.migration.datadogAgents.conversionWebhook.enabled": "true",
+				},
+			},
+			datadogAgentManifestPath: "./manifests/default.yaml",
+			operatorAssertions:       verifyOperator,
+			agentAssertions:          verifyAgent,
+			installCertManager:       true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Setup
+			namespaceName := fmt.Sprintf("datadog-agent-%s", strings.ToLower(random.UniqueId()))
+			kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+			k8s.CreateNamespace(t, kubectlOptions, namespaceName)
+			defer k8s.DeleteNamespace(t, kubectlOptions, namespaceName)
+
+			// Instal Cert Manager if needed
+			cleanupCM := setupCertManager(t, kubectlOptions, tt.installCertManager)
+			defer cleanupCM()
+
+			// Install Operator
+			cleanupOperator := common.InstallChart(t, kubectlOptions, tt.command)
+			defer cleanupOperator()
+			time.Sleep(15 * time.Second)
+			// Verify Operator
+			verifyOperator(t, kubectlOptions)
+
+			// Apply DatadogAgent Manifest
+			cleanupSecrets := common.CreateSecretFromEnv(t, kubectlOptions, apiKeyEnv, appKeyEnv)
+			defer cleanupSecrets()
+			t.Log("Applying DatadogAgent manifest")
+			k8s.KubectlApply(t, kubectlOptions, tt.datadogAgentManifestPath)
+			defer k8s.KubectlDelete(t, kubectlOptions, tt.datadogAgentManifestPath)
+
+			// Verify Agent Setup
+			t.Log("Verifying agent pods are running")
+			verifyAgent(t, kubectlOptions)
+
+			// 'Pause' test for local debugging
+			//t.Log("Sleeping for 2 minutes")
+			//time.Sleep(120 * time.Second)
+		})
+	}
+
+}
+
+func verifyOperator(t *testing.T, kubectlOptions *k8s.KubectlOptions) {
+	verifyNumPodsForSelector(t, kubectlOptions, 1, "app.kubernetes.io/name=datadog-operator")
+}
+
+func verifyAgent(t *testing.T, kubectlOptions *k8s.KubectlOptions) {
+	verifyNumPodsForSelector(t, kubectlOptions, 2, "agent.datadoghq.com/component=agent")
+	verifyNumPodsForSelector(t, kubectlOptions, 1, "agent.datadoghq.com/component=cluster-agent")
+	verifyNumPodsForSelector(t, kubectlOptions, 1, "agent.datadoghq.com/component=cluster-checks-runner")
+}
+
+func verifyNumPodsForSelector(t *testing.T, kubectlOptions *k8s.KubectlOptions, numPods int, selector string) {
+	t.Log("Waiting for number of pods created", "number", numPods, "selector", selector)
+	k8s.WaitUntilNumPodsCreated(t, kubectlOptions, v1.ListOptions{
```

Comment:
this is pretty cool, waiting for the pods

Reply:
Agree, terratest provides really convenient interface!